### PR TITLE
Adding Carthage Support

### PR DESCRIPTION
**Pull request by soheilbm**
- Adding Cartfile with Payboy and PureLayout
- Updating Tabman workspace to link those 2 libraries
- Updating Travis file to with carthage script
- Updating Readme
- Removing Podfile for Example project and just relying on compiled libraries

### CODE REVIEW

**Comment by sirvon** [comment]
@soheilbm hey i tried to build this and pageboy via carthage and im getting this error

```
ProcessInfoPlistFile /Users/R.Thomas/Library/Developer/Xcode/DerivedData/Tabman-foracmzvgkrlzxhgkgejeampktkl/Build/Products/Debug-iphoneos/Tabman.framework/Info.plist Tabman/Info.plist
    cd /Users/Shared/xna/frontend/apps/ios/LoQooFood/LoQooFood/Carthage/Checkouts/Tabman/Sources
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/sbin:/Users/R.Thomas/.rbenv/shims:/Users/R.Thomas/.rbenv/shims:/Users/R.Thomas/.rbenv/bin:/Users/R.Thomas/.jenv/shims:/usr/local/rvm/gems/ruby-2.2.2/bin:/usr/local/rvm/gems/ruby-2.2.2@global/bin:/usr/local/rvm/rubies/ruby-2.2.2/bin:/usr/bin:/usr/local/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/rvm/bin:/Users/R.Thomas/golang/bin:/Users/R.Thomas/.gem/bin:/Users/R.Thomas/Library/Android/sdk/platform-tools:/usr/local/node-v0.10.33-linux-x64/bin:/home/R.Thomas/dotfiles:/home/R.Thomas/SDK/android-studio/bin:/usr/share/dse/bin:/Users/R.Thomas/.gem/bin:/Users/R.Thomas/Library/Android/sdk/platform-tools:/usr/local:/Users/R.Thomas/.gem/bin:/Users/R.Thomas/golang/bin:/usr/local/opt/go/libexec/bin:/usr/local/ec2/ec2-api-tools-1.7.1.0/bin:/home/R.Thomas/SDK/google_appengine"
    builtin-infoPlistUtility /Users/Shared/xna/frontend/apps/ios/LoQooFood/LoQooFood/Carthage/Checkouts/Tabman/Sources/Tabman/Info.plist -expandbuildsettings -format binary -platform iphoneos -o /Users/R.Thomas/Library/Developer/Xcode/DerivedData/Tabman-foracmzvgkrlzxhgkgejeampktkl/Build/Products/Debug-iphoneos/Tabman.framework/Info.plist

PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/R.Thomas/Library/Developer/Xcode/DerivedData/Tabman-foracmzvgkrlzxhgkgejeampktkl/Build/Intermediates/Tabman.build/Debug-iphoneos/Tabman.build/Script-CD99F4CA7550180E33FC9BA7.sh
    cd /Users/Shared/xna/frontend/apps/ios/LoQooFood/LoQooFood/Carthage/Checkouts/Tabman/Sources
    /bin/sh -c /Users/R.Thomas/Library/Developer/Xcode/DerivedData/Tabman-foracmzvgkrlzxhgkgejeampktkl/Build/Intermediates/Tabman.build/Debug-iphoneos/Tabman.build/Script-CD99F4CA7550180E33FC9BA7.sh
diff: /Podfile.lock: No such file or directory


diff: /Manifest.lock: No such file or directory
error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.

** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/R.Thomas/Library/Developer/Xcode/DerivedData/Tabman-foracmzvgkrlzxhgkgejeampktkl/Build/Intermediates/Tabman.build/Debug-iphoneos/Tabman.build/Script-CD99F4CA7550180E33FC9BA7.sh
(1 failure)
A shell task (/usr/bin/xcrun xcodebuild -workspace /Users/Shared/xna/frontend/apps/ios/LoQooFood/LoQooFood/Carthage/Checkouts/Tabman/Tabman.xcworkspace -scheme Tabman -configuration Debug -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
2017-03-26 23:40:44.038 xcodebuild[2811:491341] [MT] PluginLoading: Required plug-in compatibility UUID E0A62D1F-3C18-4D74-BFE5-A4167D643966 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Alcatraz.xcplugin' not present in DVTPlugInCompatibilityUUIDs
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/R.Thomas/Library/Developer/Xcode/DerivedData/Tabman-foracmzvgkrlzxhgkgejeampktkl/Build/Intermediates/Tabman.build/Debug-iphoneos/Tabman.build/Script-CD99F4CA7550180E33FC9BA7.sh
(1 failure)
```

**Comment by soheilbm** [comment]
@sirvon did you try carthage from my branch? cuz I'm pretty sure I removed checking manifest in main file

**Comment by sirvon** [comment]
@soheilbm this is what I have in my Cartfile...
```dsl
 1 github "soheilbm/Tabman"
```
and this is the error
```bash
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/R.Thomas/Library/Developer/Xcode/DerivedData/Tabman-foracmzvgkrlzxhgkgejeampktkl/Build/Intermediates/Tabman.build/Release-iphoneos/Tabman.build/Script-CD99F4CA7550180E33FC9BA7.sh
(1 failure)
error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
A shell task (/usr/bin/xcrun xcodebuild -workspace /Users/Shared/xna/frontend/apps/ios/LoQooFood/LoQooFood/Carthage/Checkouts/Tabman/Tabman.xcworkspace -scheme Tabman -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
2017-03-27 07:45:46.468 xcodebuild[26201:1566616] [MT] PluginLoading: Required plug-in compatibility UUID E0A62D1F-3C18-4D74-BFE5-A4167D643966 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Alcatraz.xcplugin' not present in DVTPlugInCompatibilityUUIDs
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/R.Thomas/Library/Developer/Xcode/DerivedData/Tabman-foracmzvgkrlzxhgkgejeampktkl/Build/Intermediates/Tabman.build/Release-iphoneos/Tabman.build/Script-CD99F4CA7550180E33FC9BA7.sh
(1 failure)
```


I ran carthage with..."carthage update" and "carthage bootstrap --platform iOS"

**Comment by sirvon** [comment]
this worked!

```bash
git "https://github.com/soheilbm/Tabman.git" "carthage"
```

```bash
rm -rf Carthage/Checkouts/ && carthage update --platform iOS
```

File: Podfile [hunk 1]
```diff
@@ -4,22 +4,10 @@ workspace 'Tabman'
...
-end
-
 target 'Tabman-Example' do
   project './Example/Tabman-Example.xcodeproj'
   workspace 'Tabman'
 
-  shared_pods
+  pod 'Pageboy', '~> 0.4.0'
+  pod 'PureLayout', '~> 3.0.0'
 end
```
Comment (msaps) [inline Podfile]: Is there a reason for this (Not too familiar with Carthage)? Having checked out your branch and run `pod install` the project doesn't seem to be build-able?

File: README.md [hunk 2]
```diff
@@ -29,6 +32,20 @@ pod 'Tabman'
...
 
+### Carthage
+Pageboy is available through [Carthage](https://github.com/Carthage/Carthage). Simply install carthage with [Homebrew](http://brew.sh/) using the following command:
+
+```bash
+$ brew update
+$ brew install carthage
+```
+
+Add Pageboy to your `Cartfile`:
```
Comment (msaps) [inline README.md]: Pageboy referenced when it should be Tabman.

File: Cartfile.resolved [hunk 3]
```diff
@@ -0,0 +1,2 @@
+github "PureLayout/PureLayout" "v3.0.2"
+github "uias/Pageboy" "0.4.12"
```
Comment (msaps) [inline Cartfile.resolved]: I think this file needs removing from a previous merge conflict.

**Comment by msaps** [comment]
Thanks for this 👍  Just done a quick review, a few things need addressing before merging I think.

**Comment by sirvon** [comment]
@soheilbm any hints/info you lend on this [issue](https://github.com/uias/Tabman/issues/32)...

**Comment by soheilbm** [comment]
Sorry I was quite busy lately. @sirvon I look into the issue later today. 
@msaps thank you for your feedback, I'll update this PR later this week. Meanwhile, I was wondering If I could remove podfile for `Example` project because I can just easily link the framework without the needs of podfile in there!

**Comment by msaps** [comment]
@soheilbm no worries, that'd be great! 

Regarding the Podfile/Carthage, would this mean I'd be able to remove any step following checkout / before building the project? Currently you have to run `pod install` to get the external dependencies. As I understand it, Carthage would require a similar command unless the dependencies can be bundled for checkout?

Note that whatever the solution, both the Example & Tabman projects have to be buildable for CI.

**Comment by msaps** [comment]
@soheilbm obviously needs updating with the latest release from master as well. 👍

**Comment by soheilbm** [comment]
@sirvon would you mind to try again with this branch? just wanna make sure https://github.com/uias/Tabman/issues/32 is also solved

**Comment by sirvon** [comment]
```swift
Undefined symbols for architecture x86_64:
  "direct field offset for Tabman.TabmanViewController.
(bar.storage in _57F68DA9D23EDB616EF841D0392909A0) : Tabman.TabmanBarConfig?",
referenced from:
      Tabman.TabmanViewController.bar.setter : Tabman.TabmanBarConfig 
in TrendingEventsController.o
      Tabman.TabmanViewController.(bar.materializeForSet : Tabman.TabmanBarConfig).
(closure #1) in TrendingEventsController.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

```swift
//
//  UpdatesTableViewController.swift
//
//

import UIKit
import RealmSwift
import Tabman
import Pageboy

class UpdateFeedController: TabmanViewController, PageboyViewControllerDataSource {
    
    let realm = RealmManager.shared

        override func viewWillAppear(_ animated: Bool) {
            print("viewWillAppear")

        }
    
        override func viewDidLoad() {
            super.viewDidLoad()
            print("viewDidLoad")
            self.dataSource = self
            DispatchQueue.global(qos: .background).async {}
            //let realm = RealmManager.shared.readOnlyAppData.apDestinations()
            // Read some data from the bundled Realm
        }
    
        override func viewDidLayoutSubviews() {
            super.viewDidLayoutSubviews()
        }
        
        override func didReceiveMemoryWarning() {
            super.didReceiveMemoryWarning()
            // Dispose of any resources that can be recreated.
        }
    
    
    // MARK: PageboyViewControllerDataSource

    func viewControllers(forPageboyViewController pageboyViewController: PageboyViewController) -> [UIViewController]? {
        // return array of view controllers
        let viewControllerOne = ViewControllerOne()
        let viewControllerTwo = ViewControllerTwo()
        let viewControllers = [viewControllerOne, viewControllerTwo]
        
        // configure the bar
        self.bar.items = [TabmanBarItem(title: "Page 1"),
                          TabmanBarItem(title: "Page 2")]
        
        return viewControllers
    }
    
    func defaultPageIndex(forPageboyViewController pageboyViewController: PageboyViewController) -> PageboyViewController.PageIndex? {
        // use default index
        return nil
    }
    
        
    }

``

**Comment by sirvon** [comment]
hey any movement on this @soheilbm ?

**Comment by soheilbm** [comment]
@sirvon honestly I have no idea why you are seeing that error. Even this project example is using carthage now, but i don't see any errors there.

**Comment by msaps** [comment]
@soheilbm could you resolve the conflict with the Podfile.lock please?

**Comment by soheilbm** [comment]
@msaps I removed Podfile and Podfile.lock from project, since It's not needed as the project directly uses built frameworks.

**Comment by msaps** [comment]
@soheilbm thanks for all this, I'll get it in the next release 👍

**Pull request merged**